File: ee-mdd-example/src-gen/main/js/um/TaskStatus.js

function TaskStatusLit(name, code) {
  this.name = name;
  this.code = code;
}
  
TaskStatusLit.prototype = {
  constructor: TaskStatusLit,
  isUnknown() : function() {
    return this === TaskStatus.UNKNOWN; 
  },
  
  isOpen() : function() {
    return this === TaskStatus.OPEN; 
  },
  
  isClosed() : function() {
    return this === TaskStatus.CLOSED; 
  }
}

var TaskStatus = {
  UNKNOWN: new TaskStatusLit('Unknown', -1),
  OPEN: new TaskStatusLit('Open', 1),
  CLOSED: new TaskStatusLit('Closed', 2)
}

File: ee-mdd-example/src-gen/main/java/com/siemens/ra/cg/pl/um/TaskStatus.java
package com.siemens.ra.cg.pl.um;

public enum TaskStatus {
  UNKNOWN(-1),
  OPEN(1),
  CLOSED(2);
  
  protected int code; 

  private TaskStatus(int code) {
      this.code = code;
    }
  
  public int getCode() {
    return code; 
  }
  
  public boolean isUnknown() {
    return this == UNKNOWN; 
  }
  
  public boolean isOpen() {
    return this == OPEN; 
  }
  
  public boolean isClosed() {
    return this == CLOSED; 
  }
}